src/conversion/modelconversiondispatcher.js

```diff
@@ -250,7 +252,9 @@ export default class ModelConversionDispatcher {
 				item: item
 			};
 
-			this._testAndFire( 'remove', data, consumable );
+			const name = removeAs || item.name || '$text';
+
+			this._testAndFire( `remove`, consumable, data, name );
```

Comment:
I don't like the fact that you fire `data.item`, which have the name, and  `name` which usually is the same, but developer need to know that in the case of rename it can be different. It does simplify nothing because developer still needs to know about rename and now it's hidden inside `remove`.

Since the `item` in `remove` is not in the document anyway (it's in the graveyard, so have no context), can't we pass a clone of the element with the old name in case of rename as item?